modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1281,6 +1281,83 @@ CV__DNN_INLINE_NS_BEGIN
                              float confThreshold = 0.5f, float nmsThreshold = 0.0f);
      };
 
+     /** @brief This class represents high-level API for text recognition networks.
+      *
+      * SceneTextRecModel allows to set params for preprocessing input image.
+      * SceneTextRecModel creates net from file with trained weights and config,
+      * sets preprocessing input, runs forward pass and return recognition result.
+      * For TextRecognitionModel, CRNN-CTC is supported.
+      */
+     class CV_EXPORTS_W TextRecognitionModel : public Model
+     {
+     public:
+         /**
+          * @brief Create text recognition model from network represented in one of the supported formats.
+          * An order of @p model and @p config arguments does not matter.
+          * @param[in] model Binary file contains trained weights.
+          * @param[in] config Text file contains network configuration.
+          */
+         CV_WRAP TextRecognitionModel(const String& model, const String& config = "");
+
+         /**
+          * @brief Create model from deep learning network.
+          * @param[in] network Net object.
+          */
+         CV_WRAP TextRecognitionModel(const Net& network);
+
+         /**
+          * @brief Given the @p input frame, create input blob, run net and return recognition result.
+          * @param[in] frame: The input image.
+          * @param[in] decodeType: The decoding method of translating the network output into string.
```

Comment:
Which are possible values?

Reply:
It only supports "CTC-greedy" now, and I will add more decoding methods in the future, such as Beam Searching.

Reply:
User should know the options so it must be documented or we can add enum